src/libraries/apiclient/apiclientcore.js

```diff
@@ -174,6 +176,20 @@ define(["events", "appStorage"], function(events, appStorage) {
             }, 0)
         }
     }
+    
+    function socketHeartBeat(socket) {
+        // check if socket already define or not
+        if(!socket) return;
+        // ensure that socket already in the readyState
+        if(socket.readyState !== WebSocket.OPEN) return;
+        // sent heartbeat to websocket.
+        socket.send("heartbeat");
+        // set interval to 60s, because cloudflare will going to close the connection
+        // without any message being sent in 100s (for free-tier).
+        setTimeout(function() {
+            socketHeartBeat(socket);
+        }, 60000);
```

Comment:
```suggestion
        }, 20000);
```
I think I read @mcarlton00 reporting 30 seconds being timed out somewhere else, and sending a heartbeat each 20 seconds should not consume much of resources anyway :)

Reply:
You might be thinking of something else.  I was having issues in clients where they haven't "checked in" in ~15 minutes, the connection drops.

Reply:
@mcarlton00 
Means, no message being passed on the WS in 15 mins?
It probably happens also, because at the end it will depends on a lot of implementation and client (like firefox will be different with chrome), and also whether you behind reverse proxy or not (like your caddy or nginx configuration?)

@JustAMan 
actually we don't even need to sent strings, just like the actual ping/pong (which only change the control frame op code).
But it's just easier to check on the developer tools when I can see the "heartbeat" message.

Reply:
I think more context for mine would be helpful, sorry about that.

So the issue I've run into specifically is with the Mopidy extension.  If you just play shorter songs, it's all good.  But if you play something longer like an audiobook, it will report it's started playing to the server and then just continue going locally.  Currently I don't have it periodically posting updates, so the server receives a "client is playing" message and then doesn't hear anything else from the client.  After 15 minutes or so the server effectively says "It's probably not playing anymore" and stops tracking progress.  This is useful as a timeout measure if a client crashes, it doesn't mark a whole file as watched.  It just means that during playback clients need to post status updates regularly.

TLDR: the 60000 is likely fine.